src/gpu/shaderFunctions.js

```diff
@@ -273,4 +273,133 @@ bool bvhIntersectFirstHit(
 
 }
 
+
+
+
+
+
+float dot2( in vec3 v ) {
+
+	return dot(v,v);
+
+}
+
+float distSquared( vec3 a, vec3 b ) {
+
+	vec3 c = a - b;
+	return dot(c, c);
+
+}
+
+float udTriangle( vec3 p, vec3 a, vec3 b, vec3 c ) {
+
+	vec3 ba = b - a; vec3 pa = p - a;
+	vec3 cb = c - b; vec3 pb = p - b;
+	vec3 ac = a - c; vec3 pc = p - c;
+	vec3 nor = cross( ba, ac );
+	return
+		(
+			sign( dot( cross( ba, nor ), pa ) ) +
+			sign( dot( cross( cb, nor ), pb ) ) +
+			sign( dot( cross( ac, nor ), pc ) ) < 2.0
+		) ? (
+			min( min(
+			dot2( ba * clamp( dot( ba, pa ) / dot2( ba ), 0.0, 1.0 ) - pa ),
+			dot2( cb * clamp( dot( cb, pb ) / dot2( cb ), 0.0, 1.0 ) - pb ) ),
+			dot2( ac * clamp( dot( ac, pc ) / dot2( ac ), 0.0, 1.0 ) - pc) )
+		) : (
+			dot( nor, pa ) * dot( nor, pa ) / dot2( nor )
+		);
+
+}
```

Comment:
cc @n8python - where did this function come from? Do you have a reference for the math here?

Reply:
Just a simple distance to triangle function!
https://www.shadertoy.com/view/4sXXRN
From our lord and savior inigo quilez!